lib/mongoid/extensions/big_decimal.rb

```diff
@@ -46,7 +50,13 @@ module ClassMethods
         #
         # @return [ BigDecimal, nil ] A BigDecimal derived from the object or nil.
         def demongoize(object)
-          object && object.numeric? ? BigDecimal(object.to_s) : nil
+          unless object.nil?
+            if object.is_a?(BSON::Decimal128)
+              object.to_big_decimal
+            elsif object.numeric?
+              BigDecimal(object.to_s)
+            end
+          end
         end
 
         # Mongoize an object of any type to how it's stored in the db as a String.
```

Comment:
I think this needs to be updated, specifically the `as a String` part?

Reply:
done